crypto/ec/ec_kmeth.c

```diff
@@ -166,6 +166,12 @@ EC_KEY *EC_KEY_new_method(ENGINE *engine)
     ret->references = 1;
 
     if (ret->meth->init != NULL && ret->meth->init(ret) == 0) {
+        ECerr(EC_F_EC_KEY_NEW_METHOD, ERR_R_INIT_FAIL);
+#ifndef OPENSSL_NO_ENGINE
+        ENGINE_finish(ret->engine);
```

Comment:
This change is redoing the work of  EC_KEY_free().
So it creates a double-free issue, three times.
---
IMHO, this method could be simplied as I do in another PR 
See  e178c76fe9e4a197b9dab931451c57c340a25a8e